proposals/csharp-10.0/improved-interpolated-strings.md

```diff
@@ -291,10 +291,69 @@ as a _member\_access_ through type `T`.
     * If no single-best method was found, the result of overload resolution is ambiguous, an error is produced, and no further steps are taken.
 4. Final validation on `F` is performed.
     * If any element of `A` occurred lexically after `i`, an error is produced and no further steps are taken.
+    * If any `A` requests the receiver of `F`, and `F` is an indexer being used as an _initializer\_target_ in a _member\_initializer_, then an error is reported and no further steps are taken.
 
 Note: the resolution here intentionally do _not_ use the actual expressions passed as other arguments for `Argx` elements. We only consider the types post-conversion. This makes sure that we
 don't have double-conversion issues, or unexpected cases where a lambda is bound to one delegate type when passed to `M1` and bound to a different delegate type when passed to `M`.
 
+Note: We report an error for indexers uses as member initializers because of the order of evaluation for nested member initializers. Consider this code snippet:
+
+```cs
+
+var x1 = new C1 { C2 = { [GetString()] = { A = 2, B = 4 } } };
+
+/* Lowering:
+__c1 = new C1();
+CustomHandler argTemp = GetString();
```

Comment:
Is `CustomHandler` right in this example, or is this meant to represent the simple/string scenario?

Reply:
You didn't see anything.